prep_sector_split.R

```diff
@@ -49,10 +53,51 @@ dotenv::load_dot_env()
 input_path_split <- file.path(Sys.getenv("DIR_SPLIT_COMPANY_ID"), Sys.getenv("FILENAME_SPLIT_COMPANY_ID"))
 input_path_advanced_company_indicators <- file.path(Sys.getenv("DIR_ADVANCED_COMPANY_INDICATORS"), Sys.getenv("FILENAME_ADVANCED_COMPANY_INDICATORS"))
 input_path_matched <- Sys.getenv("DIR_MATCHED")
+input_path_company_metrics <- file.path(Sys.getenv("DIR_OUTPUT"), "equal_weights", "aggregated")
+# output_path_aggregated <- file.path(output_path, "aggregated")
 
 # start_year should match the year of the ABCD data release
 start_year <- as.numeric(Sys.getenv("PARAM_START_YEAR"))
+time_frame_select <- as.integer(Sys.getenv("PARAM_TIME_FRAME"))
 
+# generate worst case sector split----
+# NOTE: this requires having run the script run_aggregate_loanbooks.R, as the
+# generation of the worst case sector split depends on the availability of
+# company level results for all sectors.
+# This can be obtained, e.g., by running the analysis using equal weights.
+if (dir.exists(input_path_company_metrics)) {
+  ## load input data----
+  # generate worst case sector splits
+  company_aggregated_alignment_net_tms <-
+    readr::read_csv(file.path(input_path_company_metrics, "company_aggregated_alignment_net_tms.csv"))
+
+  company_aggregated_alignment_net_sda <-
+    readr::read_csv(file.path(input_path_company_metrics, "company_aggregated_alignment_net_sda.csv"))
+
+  company_aggregated_alignment_net <- company_aggregated_alignment_net_tms %>%
+    dplyr::bind_rows(company_aggregated_alignment_net_sda)
+
+  ## calculate sector split----
+
+  companies_sector_split_worst_case <- company_aggregated_alignment_net %>%
+    dplyr::filter(.data$year == .env$start_year + .env$time_frame_select) %>%
+    dplyr::group_by(
+      .data$group_id, .data$name_abcd, .data$region, .data$scenario_source, .data$scenario, .data$direction
+    ) %>%
+    dplyr::slice_min(.data$alignment_metric) %>%
+    dplyr::ungroup() %>%
+    dplyr::distinct(.data$name_abcd, .data$sector) %>%
+    dplyr::mutate(sector_split = 1) %>%
+    dplyr::rename(name_company = "name_abcd")
```

Comment:
General comment (not necessary for this PR), but I guess this could be a good candidate for extraction into a "worst_cast_split" generating package function

Reply:
I think mid term this repo has a lot of pieces to disentangle to be honest. But the degree to which that should be done, to me, depends a bit on what we define as product targets for P4B/P4S in the mid term. But I agree with your take on this